plugins/db-sqlserver/src/index.ts

```diff
@@ -0,0 +1,315 @@
+          field.customAttributes || undefined 
+        ),
+        // Prisma Scalar Relation Field
+        PrismaSchemaDSL.createScalarField(
+          scalarRelationFieldName,
+          MSSQLServerPlugin.idTypeToPrismaScalarType[idType],
+          false,
+          field.required,
+          !field.properties.allowMultipleSelection &&
+            !relatedField?.properties.allowMultipleSelection &&
+            !isOneToOneWithoutForeignKey
+            ? true
+            : field.unique,
+          undefined,
+          undefined,
+          undefined,
+          undefined,
+          undefined,
+          field.customAttributes || undefined
+        ),
+      ];
+    };
+
+    return {
+      ...eventParams,
+      dataSource: dataSource,
+    };
+  }
+
+  /**
+   * Creates Prisma Schema relation name according to the names of the entity,
+   * field, relatedEntity and relatedField.
+   * This function is assumed to be used when a relation name is necessary
+   * @param entity
+   * @param field
+   * @param relatedEntity
+   * @param relatedField
+   * @param fieldHasUniqueName
+   * @returns Prisma Schema relation name
+   * @todo use unique name of one of the fields deterministically (VIPCustomers or VIPOrganizations)
+   */
+  private static createRelationName(
+    entity: Entity,
+    field: EntityField,
+    relatedEntity: Entity,
+    relatedField: EntityField,
+    fieldHasUniqueName: boolean,
+    relatedFieldHasUniqueName: boolean
+  ): string {
+    const relatedEntityNames = [
+      relatedEntity.name,
+      pascalCase(relatedEntity.pluralName),
+    ];
+    const entityNames = [entity.name, pascalCase(entity.pluralName)];
+    const matchingRelatedEntityName = relatedEntityNames.find(
+      (name) => field.name === camelCase(name)
+    );
+    const matchingEntityName = entityNames.find(
+      (name) => relatedField.name === camelCase(name)
+    );
+    if (matchingRelatedEntityName && matchingEntityName) {
+      const names = [matchingRelatedEntityName, matchingEntityName];
+      // Sort names for deterministic results regardless of entity and related order
+      names.sort();
+      return names.join("On");
+    }
+    if (fieldHasUniqueName || relatedFieldHasUniqueName) {
+      const names = [];
+      if (fieldHasUniqueName) {
+        names.push(field.name);
+      }
+      if (relatedFieldHasUniqueName) {
+        names.push(relatedField.name);
+      }
+      // Sort names for deterministic results regardless of entity and related order
+      names.sort();
+      return names[0];
+    }
+    const entityAndField = [entity.name, field.name].join(" ");
+    const relatedEntityAndField = [relatedEntity.name, relatedField.name].join(
+      " "
+    );
+    const parts = [entityAndField, relatedEntityAndField];
+    // Sort parts for deterministic results regardless of entity and related order
+    parts.sort();
+    return pascalCase(parts.join(" "));
+  }
+
+  private static idTypeToPrismaScalarType: {
```

Comment:
It might be a better idea to move this mapper on the `code-gen-types` otherwise we will need a plugin release for any new idType